### PR TITLE
Move web3 event handlers into global

### PR DESCRIPTION
**Pull request by crabsinger**
We want the web3 event handlers to be setup globally, not just via the auth modal, so moving these into Web3React hook setup.

Additionally adding a reload on accountsChanged hook.

### CODE REVIEW

File: src/hooks/useWeb3React.tsx [hunk 1]
```diff
@@ -81,3 +84,30 @@ function getLibrary(provider: any): Web3Provider {
...
+        // When a Provider makes its initial connection, it emits a "network"
+        // event with a null oldNetwork along with the newNetwork. So, if the
+        // oldNetwork exists, it represents a changing network
+        if (oldNetwork) {
+          window.location.reload();
+        }
+      });
+
+      // Web3Provider doesn't work with accountsChanged
+      ethereum.on('accountsChanged', () => {
```
Comment (crabsinger) [inline src/hooks/useWeb3React.tsx:105]: @topocount not sure why but using `provider.on` doesn't work here. Any idea why the Ethers provider wouldn't pass through this event? But using the plain object does work?

Comment (topocount) [reply src/hooks/useWeb3React.tsx:105]: @crabsinger I'm not sure what

> using the plain object does work

means. Could you sketch the working example?

Comment (crabsinger) [reply src/hooks/useWeb3React.tsx:105]: `ethereum.on('accountsChanged', ()` works
`provider.on('network',` does not work.

@topocount does that make sense? Wrapping the `window.ethereum` object in a Ethers Web3Provider breaks the functionality.

Comment (crabsinger) [reply src/hooks/useWeb3React.tsx:105]: https://github.com/ethers-io/ethers.js/issues/1396#issuecomment-806380431 answer here!

**Review by levity — DISMISSED** [review]

**Review by levity — APPROVED** [review]

**Pull request merged**